diff --git a/hledger-lib/Text/Megaparsec/Custom.hs b/hledger-lib/Text/Megaparsec/Custom.hs
index 0d217f2f..f5ef950a 100644
--- a/Text/Megaparsec/Custom.hs
+++ b/Text/Megaparsec/Custom.hs
@@ -176,7 +172,7 @@ reparseExcerpt (SourceExcerpt offset txt) p = do
     Left errBundle -> customFailure $ ErrorReparsing $ bundleErrors errBundle
 
   where
-    offsetInitialState :: Int -> s -> State s
+    offsetInitialState :: Int -> s -> State s e
     offsetInitialState initialOffset s = State
       { stateInput  = s
       , stateOffset = initialOffset
@@ -187,6 +183,7 @@ reparseExcerpt (SourceExcerpt offset txt) p = do
         , pstateTabWidth = defaultTabWidth
         , pstateLinePrefix = ""
         }
+      , stateParseErrors = []
       }
 
 --- * Pretty-printing custom parse errors
