diff --git a/src/Idris/Parser/Stack.hs b/src/Idris/Parser/Stack.hs
index fb7b61144..991f50d48 100644
--- a/src/Idris/Parser/Stack.hs
+++ b/src/Idris/Parser/Stack.hs
@@ -84,7 +84,7 @@ instance Message ParseError where
       (pos, _) = P.reachOffsetNoLine (parseErrorOffset err) (parseErrorPosState err)
 #endif
   messageText = PP.text . init . P.parseErrorTextPretty . parseError
-  messageSource err = Just sline
+  messageSource err = sline
     where
 #if MIN_VERSION_megaparsec(8,0,0)
       (sline, _) = P.reachOffset (parseErrorOffset err) (parseErrorPosState err)
