diff --git a/modern-uri.cabal b/modern-uri.cabal
index 9397df9..95217fb 100644
--- a/modern-uri.cabal
+++ b/modern-uri.cabal
@@ -32,7 +32,7 @@ library
                     , contravariant    >= 1.3 && < 2.0
                     , deepseq          >= 1.3 && < 1.5
                     , exceptions       >= 0.6 && < 0.11
-                    , megaparsec       >= 7.0 && < 8.0
+                    , megaparsec       >= 7.0 && < 9.0
                     , mtl              >= 2.0 && < 3.0
                     , profunctors      >= 5.2.1 && < 6.0
                     , reflection       >= 2.0 && < 3.0
@@ -67,7 +67,7 @@ test-suite tests
                     , bytestring       >= 0.2 && < 0.11
                     , hspec            >= 2.0 && < 3.0
                     , hspec-megaparsec >= 2.0 && < 3.0
-                    , megaparsec       >= 7.0 && < 8.0
+                    , megaparsec       >= 7.0 && < 9.0
                     , modern-uri
                     , text             >= 0.2 && < 1.3
   build-tools:        hspec-discover   >= 2.0 && < 3.0
@@ -86,7 +86,7 @@ benchmark bench-speed
   build-depends:      base             >= 4.11 && < 5.0
                     , bytestring       >= 0.2 && < 0.11
                     , criterion        >= 0.6.2.1 && < 1.6
-                    , megaparsec       >= 7.0 && < 8.0
+                    , megaparsec       >= 7.0 && < 9.0
                     , modern-uri
                     , text             >= 0.2  && < 1.3
   if flag(dev)
@@ -102,7 +102,7 @@ benchmark bench-memory
   build-depends:      base             >= 4.11 && < 5.0
                     , bytestring       >= 0.2 && < 0.11
                     , deepseq          >= 1.3 && < 1.5
-                    , megaparsec       >= 7.0 && < 8.0
+                    , megaparsec       >= 7.0 && < 9.0
                     , modern-uri
                     , text             >= 0.2  && < 1.3
                     , weigh            >= 0.0.4
