diff --git a/parser-combinators-tests/tests/Control/Applicative/PermutationsSpec.hs b/parser-combinators-tests/tests/Control/Applicative/PermutationsSpec.hs
index 96ed432..82216f8 100644
--- a/tests/Control/Applicative/PermutationsSpec.hs
+++ b/tests/Control/Applicative/PermutationsSpec.hs
@@ -91,7 +91,7 @@ prsp p = prs (runPermutation p)
 prsp'
   :: Permutation Parser a
   -> String
-  -> (State String, Either (ParseErrorBundle String Void) a)
+  -> (State String Void, Either (ParseErrorBundle String Void) a)
 prsp' p = prs' (runPermutation p)
 
 testPermParser :: Permutation Parser String
